src/App.js

```diff
@@ -1,19 +1,141 @@
 import React from 'react';
 import './App.css';
+import { TodoList } from './TodoList';
 
 import users from './api/users';
+import todosFromServer from './api/todos';
 
-function App() {
-  return (
-    <div className="App">
-      <h1>Add todo form</h1>
-
-      <p>
-        <span>Users: </span>
-        {users.length}
-      </p>
-    </div>
-  );
+const getUserById = userId => users.find(user => user.id === userId);
+
+const todosWithUser = todosFromServer.map(todo => ({
+  ...todo,
+  user: getUserById(todo.userId),
+}));
+
+class App extends React.Component {
+  state = {
+    todos: todosWithUser,
+    newTodoTitle: '',
+    newTodoUserId: 0,
+    hasTitleError: false,
+    hasUserError: false,
+  }
+
+  hendelFormSubmit = (event) => {
+    event.preventDefault();
+
+    const { newTodoTitle, newTodoUserId } = this.state;
+
+    this.setState({
+      hasTitleError: !newTodoTitle,
+      hasUserError: !newTodoUserId,
+    });
+
+    if (!newTodoTitle) {
+      return;
+    }
+
+    if (!newTodoUserId) {
+      return;
+    }
+
+    this.addTodo(newTodoTitle, newTodoUserId);
+  };
+
+  hendelTitleChange = (event) => {
+    this.setState({
+      newTodoTitle: event.target.value,
```

Comment:
```suggestion
      newTodoTitle: event.target.value,
      hasTitleError: false,
```
Тут можно добавить, чтобы ошибка скидывалась при вводе

Reply:
Yep. Task description says that error should be not visible when the user starts to enter the title:

> hide a message immediately after any change of the field with an error